src/Utilities/Compiler/Options/AbstractCategorizedAnalyzerConfigOptions.cs

```diff
@@ -88,13 +89,23 @@ public bool TryGetOptionValue<T>(string optionName, OptionKind kind, DiagnosticD
             }
         }
 
-        private (bool found, object? value) ComputeOptionValue<T>(string optionName, OptionKind kind, DiagnosticDescriptor rule, TryParseValue<T> tryParseValue)
+        private (bool found, object? value) ComputeOptionValue<T>(string optionName, OptionKind kind, DiagnosticDescriptor? rule, TryParseValue<T> tryParseValue)
         {
             var optionKeyPrefix = MapOptionKindToKeyPrefix(kind);
-            if (TryGetSpecificOptionValue(rule.Id, optionKeyPrefix, out var optionValue) ||
-                TryGetSpecificOptionValue(rule.Category, optionKeyPrefix, out optionValue) ||
-                TryGetAnySpecificOptionValue(rule.CustomTags, optionKeyPrefix, out optionValue) ||
-                TryGetGeneralOptionValue(optionKeyPrefix, out optionValue))
+
+            T optionValue;
+            if (rule != null)
+            {
+                if (TryGetSpecificOptionValue(rule.Id, optionKeyPrefix, out optionValue) ||
```

Comment:
[suggestion] could be merged with outer if.

Reply:
That required adding parentheses around the second checks and was making the code clumsy :-)